src/ovs/aca_ovs_l3_programmer.cpp

```diff
@@ -42,49 +42,346 @@ ACA_OVS_L3_Programmer &ACA_OVS_L3_Programmer::get_instance()
+    for (int i = 0; i < current_RouterConfiguration.subnet_routing_tables_size(); i++) {
+      auto current_subnet_routing_table =
+              current_RouterConfiguration.subnet_routing_tables(i);
+
+      string current_router_subnet_id = current_subnet_routing_table.subnet_id();
+
+      ACA_LOG_DEBUG("Processing subnet ID: %s for router ID: %s.\n",
+                    current_router_subnet_id.c_str(),
+                    current_RouterConfiguration.id().c_str());
+
+      // check if current_router_subnet_id already exist in new_subnet_routing_tables
+      if (new_subnet_routing_tables.find(current_router_subnet_id) !=
+          new_subnet_routing_tables.end()) {
+        is_subnet_routing_table_exist = true;
+      }
+
+      // Look up the subnet configuration to query for additional info
+      for (int j = 0; j < parsed_struct.subnet_states_size(); j++) {
+        SubnetConfiguration current_SubnetConfiguration =
+                parsed_struct.subnet_states(j).configuration();
+
+        ACA_LOG_DEBUG("current_SubnetConfiguration subnet ID: %s.\n",
+                      current_SubnetConfiguration.id().c_str());
+
+        if ((parsed_struct.subnet_states(j).operation_type() == OperationType::INFO) &&
+            (current_SubnetConfiguration.id() == current_router_subnet_id)) {
+          found_vpc_id = current_SubnetConfiguration.vpc_id();
+
+          found_cidr = current_SubnetConfiguration.cidr();
+
+          slash_pos = found_cidr.find('/');
+          if (slash_pos == string::npos) {
+            throw std::invalid_argument("'/' not found in cidr");
+          }
+
+          found_tunnel_id = current_SubnetConfiguration.tunnel_id();
+          if (!aca_validate_tunnel_id(found_tunnel_id)) {
+            throw std::invalid_argument("found_tunnel_id is invalid");
+          }
+
+          // subnet info's gateway ip and mac needs to be there and valid
+          found_gateway_ip = current_SubnetConfiguration.gateway().ip_address();
+
+          // inet_pton returns 1 for success 0 for failure
+          if (inet_pton(AF_INET, found_gateway_ip.c_str(), &(sa.sin_addr)) != 1) {
+            throw std::invalid_argument("found gateway ip address is not in the expect format");
+          }
+
+          found_gateway_mac = current_SubnetConfiguration.gateway().mac_address();
+
+          if (!aca_validate_mac_address(found_gateway_mac.c_str())) {
+            throw std::invalid_argument("found_gateway_mac is invalid");
+          }
+
+          subnet_routing_table_entry new_subnet_routing_table_entry;
+
+          if (is_subnet_routing_table_exist) {
+            new_subnet_routing_table_entry =
+                    new_subnet_routing_tables[current_router_subnet_id];
+          }
+
+          // update the subnet routing table entry
+          new_subnet_routing_table_entry.vpc_id = found_vpc_id;
+          new_subnet_routing_table_entry.network_type =
+                  current_SubnetConfiguration.network_type();
```

Comment:
Do we allow update of vpcId and networkType?

Reply:
this is also unlikely, but we allow it in the router configuration logic unless @cj-chung or @xieus tell me not to :)